package.json

```diff
@@ -39,21 +39,26 @@
     "discord-oauth2": "^2.11.0",
     "dotenv": "^16.0.3",
     "dotenv-flow": "^3.2.0",
+    "envalid": "^7.3.1",
     "express": "^4.18.2",
     "express-async-router": "^0.1.15",
+    "express-response-errors": "^1.0.5",
     "jsonwebtoken": "^9.0.0",
     "lodash": "^4.17.21",
     "next": "^12.1.2",
+    "next-absolute-url": "^1.2.2",
```

Comment:
We are using nextjs, so we should be taking full advantage of its server-side-rendering capabilities. The [`getInitialProps`](https://nextjs.org/docs/api-reference/data-fetching/get-initial-props) of a page is _isomorphic_, which means that it runs on both the client and the server:
* It runs on the server when the page is loaded fresh
* It runs on the client when doing a client-side transition

This is just a [library](https://www.npmjs.com/package/next-absolute-url) to help format our data-fetching calls.